providers/google.go

```diff
@@ -124,13 +163,8 @@ func (g *GoogleOp) requestTokens(ctx context.Context, cicHash string) (*simpleoi
 	logrus.Infof("listening on http://%s/", ln.Addr().String())
 	logrus.Info("press ctrl+c to stop")
 
-	g.server = &http.Server{}
-	go func() {
-		err := g.server.Serve(ln)
-		if err != nil && err != http.ErrServerClosed {
-			logrus.Error(err)
-		}
-	}()
+	mux := http.NewServeMux()
+	g.server = &http.Server{Handler: mux}
```

Comment:
Since `g.server.Serve(ln)` now happens after `defer shutdownServer()` what happens if an error causes the function to return before running `g.server.Serve(ln)`. Will this cause a nil panic since the server isn't running or is this safe because you did initialize `g.server`?

Reply:
`g.server` is set before the `shutdownServer` closure is defined, so I don't think we'll get a nil panic even if the server isn't running (also according to `Shutdown()` it makes future calls on `Serve` return `ErrServerClosed`).